### ISSUE TITLE
Feature Request: Track Prev and Next Page Events

### ISSUE THREAD

**Issue by asap247** [ronalfy/wordpress-gravity-forms-event-tracking#11]
Hi,

First, great plugin you created.  Keep up the good work.  I have a feature request.  Would it be possible to track the multi-form prev. and next pages as events.  For example; I have a form with multiple pages.  I want to know how far people got in the form submission process  (i.e. what page did they get to before they left).  My multi-form looks like the following:

1. Main Input Page
2. Preview Submission Page
3. Pre-Cart Page
Actual Submit Button
4. Thank You Page

In other words, I want to know how many people got to the pre-cart page, preview page, etc.  Would it be possible to get this type of feature implemented.

Thank You in Advance

AJ

**Comment by nathanmarks** [comment]
This is definitely being considered. Still thinking about what the best implementation is in terms of providing the necessary options without it being too cluttered or confusing. Got any suggestions?

**Comment by asap247** [comment]
Thanks for the consideration;  I think it will be a great feature to have.  It's actually not as complicated as we think, because gravity forms has a custom hook for prev and next pages.

Gform_page_loaded

http://www.gravityhelp.com/documentation/gravity-forms/extending-gravity-forms/hooks/javascript/gform_page_loaded/

The page_loaded hook will fire when prev and next page is used.  As far as knowing which page the user got to, you could use analytics funnels through the goals setting.  Goals can be created through Google analytics, and funnels can help keep track of which prev/next page the user got to.  The problem with funnels is you currently can't use events as part of the goal funnels.  What you would have to do then is create a virtual pageview.  Then just a assign the virtual page view to the goal funnel.

Depending on the analytics code type, you can fire a virtual pageview.

```
<script type="text/javascript">// <![CDATA[ jQuery(document).bind('gform_page_loaded', function(event, form_id, current_page){ _gaq.push(['_trackPageview'], window.location.pathname + current_page); }); // ]]></script>

```

For universal analytics, you would have to use a different virtual pageview code;

ga('send', 'pageview', 'page path');

**Comment by nathanmarks** [comment]
Hey @asap247, this plugin is actually not using Javascript at all. It is leveraging the measurement protocol which allows for server to server tracking.

The measurement protocol has a **lot** of features and I've got a lot to explore and flesh out in this plugin as time goes on.

**Comment by nathanmarks** [comment]
v2 is not too far away and I'm going to hopefully introduce this as part of the refactor.
